### DISCUSSION TITLE
so-playbook-reset - can't import or have any playbook

### DISCUSSION THREAD

**Discussion by Mav1814** [Security-Onion-Solutions/securityonion#11707] [2.4] [answered]
### Version

Other (please provide detail below)

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

4

### RAM

32GB

### Storage for /

212GB

### Storage for /nsm

200GB

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hello it's me again.

After failing to create a customized playbook it simply doesn't work, the playbook runs, but no alert is generated. I tried resetting the playbooks in order to clean up and then choose the ones that i need to import again. The conclusion? 

I did so-playbook-reset and now I can't import any playbooks.
I can't run so-playbook-sigmar-efresh,  it says "State file found, exiting...
Rerun with --force to override."
when i use so-playbook-import it says "Play Creation Error - ",   
and when I try to access as administrator I get Redmine 500 error.

Has anyone encountered this problem? The documentation doesn't even mention so-playbook-reset.

It's really frustrating to always be chasing losses and not be able to get this stable.

Thanks anyway

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by Mav1814** [comment]
Any ideia?

**Comment by defensivedepth** [comment]
What exactly are you attempting to do?

**Reply by Mav1814** [reply]
Hi @defensivedepth. 

Initially I tried to create a custom playbook for windows #11499. But without success. 

The only thing I wanted was to clear out the imported/existing playbooks and import them again. Because I had imported all the available ones in order to test them.

**Reply by Mav1814** [reply]
Any help? 

I cant re-import the playbooks from rulesets, and by doing so-playbook-sigma-refresh or so-playbook-import.

**Reply by Mav1814** [reply]
Even the IDH playbooks i can't re-import.

**Comment by defensivedepth** [comment]
The only thing you should need to do to reset the Playbook toolkit is run:

`sudo so-playbook-reset` 

This can take over an hour to completely reset.

Please do this and let me know what you see.

**Reply by Mav1814** [reply]
Hi @defensivedepth :)

I've done it at least 2/3 times over the last two weeks. 

The results were the same. I always did sudo so-playbook-reset at the end of the day to give time until the next day to completely reset the playbooks. Even after that, I can't seem to import. The behavior is 

```
I can't run "so-playbook-sigmar-refresh", it says "State file found, exiting...
Rerun with --force to override."
when I use "so-playbook-import", it says "Play Creation Error - ",
and when I try to access as administrator, I get the error Redmine 500.
```
After a while, the Redmine 500 error stops and I can access the playbook as an administrator.  Even IDH playbooks are not imported.

**Reply by defensivedepth** [reply]
Neither of those are needed (so-playbook-sigma-refresh & so-playbook-import). 

All you should have to do is run so-playbook-reset and give it an hour.

After you run that command, do you see any Plays in Playbook?

**Reply by Mav1814** [reply]
I don't see anything in the Playbooks after the reset. Everything's clean (there's no play at all).
I can do it again and wait an hour.  And post the results here. 

I only did the procedure (so-playbook-sigma-refresh & so-playbook-import) because I didn't have any plays and I didn't know that, after the reset, the plays should be imported auto again.

**Reply by defensivedepth** [reply]
Yes, and any errors that you see from running so-playbook-reset

**Reply by Mav1814** [reply]
Results from the reset, now i will wait at least 2 hours. Then i will say something. 

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36708408/c783d576-0ef9-42f0-81bd-07f855f8de5a)

Do you want me to check on logs for more information?

**Reply by Mav1814** [reply]
After 2 hours, i have this results. Its the same behavior as before. I still can't login as admin on the playbook page. 

On /opt/so/log/playbook/playbook.log

```
App 114 output: I, [2023-11-13T14:07:34.673557 #114]  INFO -- : Started GET "/playbook/login" for 172.17.1.1 at 2023-11-13 14:07:34 +0000
App 114 output: I, [2023-11-13T14:07:34.675868 #114]  INFO -- : Processing by AccountController#login as HTML
App 114 output: I, [2023-11-13T14:07:34.681756 #114]  INFO -- :   Current user: anonymous
App 114 output: I, [2023-11-13T14:07:34.684331 #114]  INFO -- :   Rendering account/login.html.erb within layouts/base
App 114 output: I, [2023-11-13T14:07:34.686332 #114]  INFO -- :   Rendered account/login.html.erb within layouts/base (1.9ms)
App 114 output: I, [2023-11-13T14:07:34.699422 #114]  INFO -- :   Rendered plugins/redmine_playbook/app/views/playbook/_footer.html.erb (0.0ms)
App 114 output: I, [2023-11-13T14:07:34.699732 #114]  INFO -- : Completed 200 OK in 24ms (Views: 13.3ms | ActiveRecord: 4.5ms)
App 114 output: I, [2023-11-13T14:07:40.782586 #114]  INFO -- : Started POST "/playbook/login" for 172.17.1.1 at 2023-11-13 14:07:40 +0000
App 114 output: I, [2023-11-13T14:07:40.784009 #114]  INFO -- : Processing by AccountController#login as HTML
App 114 output: I, [2023-11-13T14:07:40.784069 #114]  INFO -- :   Parameters: {"utf8"=>"✓", "authenticity_token"=>"B9KMuEloEH4Hevq9sEG+etFbFzy/oiKyRu3u4A1MjsfHu4r7gaAcxMpgElrDX5gD7WSSYB0p6ts5a0TsMFr3DQ==", "back_url"=>"/playbook/projects/detection-playbooks/issues", "username"=>"admin", "password"=>"[FILTERED]", "login"=>"Login"}
App 114 output: I, [2023-11-13T14:07:40.788067 #114]  INFO -- :   Current user: anonymous
App 114 output: I, [2023-11-13T14:07:40.795043 #114]  INFO -- : Completed 500 Internal Server Error in 11ms (ActiveRecord: 4.5ms)
App 114 output: F, [2023-11-13T14:07:40.796045 #114] FATAL -- :   
App 114 output: F, [2023-11-13T14:07:40.796091 #114] FATAL -- : ActiveModel::MissingAttributeError (missing attribute: twofa_scheme):
App 114 output: F, [2023-11-13T14:07:40.796103 #114] FATAL -- :   
App 114 output: F, [2023-11-13T14:07:40.796117 #114] FATAL -- : app/models/user.rb:406:in `twofa_active?'
App 114 output: app/controllers/account_controller.rb:321:in `password_authentication'
App 114 output: app/controllers/account_controller.rb:307:in `authenticate_user'
App 114 output: app/controllers/account_controller.rb:41:in `login'
App 114 output: lib/redmine/sudo_mode.rb:61:in `sudo_mode'
App 114 output: I, [2023-11-13T14:07:49.338795 #114]  INFO -- : Started GET "/playbook/login" for 172.17.1.1 at 2023-11-13 14:07:49 +0000
App 114 output: I, [2023-11-13T14:07:49.339827 #114]  INFO -- : Processing by AccountController#login as HTML
App 114 output: I, [2023-11-13T14:07:49.343916 #114]  INFO -- :   Current user: anonymous
App 114 output: I, [2023-11-13T14:07:49.345680 #114]  INFO -- :   Rendering account/login.html.erb within layouts/base
App 114 output: I, [2023-11-13T14:07:49.346834 #114]  INFO -- :   Rendered account/login.html.erb within layouts/base (1.1ms)
App 114 output: I, [2023-11-13T14:07:49.365292 #114]  INFO -- :   Rendered plugins/redmine_playbook/app/views/playbook/_footer.html.erb (0.1ms)
App 114 output: I, [2023-11-13T14:07:49.365915 #114]  INFO -- : Completed 200 OK in 26ms (Views: 15.0ms | ActiveRecord: 7.0ms)
```

And on the playbooks page

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/36708408/797471a2-d1c3-4922-a949-882e0ee0349b)

I don't understand what i'm missing here.

**Reply by defensivedepth** [reply]
What version of Security Onion are you running?

**Reply by Mav1814** [reply]
I'm running the latest, 2.4.20. 

If I'm not mistaken, a new version will be released today.

**Reply by defensivedepth** [reply]
I may have been able to duplicate the issue. Digging into it further.

**Reply by Mav1814** [reply]
I'm happy to help in anyway. 
I will wait for more updates.

**Reply by defensivedepth** [reply, marked as answer]
On your Manager, open the following file:

`sudo vi /opt/so/saltstack/local/pillar/secrets.sls`

Delete the value for the `playbook_automation_api_key:` - just the value, not anything else. Save it and the run `sudo so-playbook-reset`

I was able to get it to work on my local system with that change.

**Reply by Mav1814** [reply]
Okay @defensivedepth i will do it again right now! I will let you know when its done! Thank you for everything.

**Reply by Mav1814** [reply]
Thank you! Everything is working know. 
So the problem was that the API key was adding keys, instead of removing the old one and just adding the new one, right?

**Reply by Mav1814** [reply]
Another quick questions, the plays of opencanary from the IDH didn't import. What do i need to do?

**Reply by Mav1814** [reply]
Only had to do so-playbook-import. And it's imported.

**Reply by defensivedepth** [reply]
Great! 

btw, I created an issue to track this bug: https://github.com/Security-Onion-Solutions/securityonion/issues/11790****
